pallets/ddc-staking/src/lib.rs

```diff
@@ -247,68 +255,39 @@ pub mod pallet {
 				Err(Error::<T>::AlreadyPaired)?
 			}
 
+			let stash_free = T::Currency::free_balance(&stash);
+
 			// Reject a bond which is considered to be _dust_.
-			if value < T::Currency::minimum_balance() {
+			if stash_free < T::Currency::minimum_balance() {
+				Err(Error::<T>::InsufficientBond)?
+			}
+
+			let bond_size = BondSize::<T>::get();
+
+			// Reject a bond which is lower then required.
+			if stash_free < bond_size {
```

Comment:
We check the free balance instead of the total balance to not rely on funds locked for other reasons and ensure we can always slash this amount regardless of the other liabilities of the account.